File: javascript/src/get-pis-auth-url-with-created-payee.js
import config from './config.js';
import { Moneyhub } from '@mft/moneyhub-api-client';
import yargs from 'yargs';
import { hideBin } from 'yargs/helpers';

const args = yargs(hideBin(process.argv)).option('state', {
  alias: 's',
  describe: 'state from original authorisation URL',
  default: 'foo',
}).option('nonce', {
  alias: 'n',
  describe: 'nonce for original authorisation URL',
  default: 'bar',
}).argv;

const main = async () => {
  const moneyhub = await Moneyhub(config);
  const payee = await moneyhub.addPayee({
    accountNumber: '12345678',
    name: 'payee-name',
    sortCode: '123456',
  });
  const url = await moneyhub.getPaymentAuthorizeUrl({
    state: args.state,
    amount: 1000,
    bankId: 'test',
    payeeRef: 'reference',
    payerRef: 'reference',
    payeeId: payee.data.id,
    nonce: args.nonce
  });
  console.log(url);
};

main();

File: python/src/get_pis_auth_url_with_created_payee.py
from config import *
import utils
import argparse

parser = argparse.ArgumentParser(description='Get PIS authorisation URL that creates a payee')
parser.add_argument("-s", "--state", default="foo", help="The state value")
parser.add_argument("-n", "--nonce", default="bar", help="The nonce value")
args = parser.parse_args()

identity_server = config["identity_service_url"]

payee = utils.create_payee(config, name="payee name", sort_code="123456", account_number="12345678")
payee_id = payee["data"]["id"]

client_id = config["client"]["client_id"]
params = {
  "client_id": client_id,
  "scope": "openid id:test payment",
  "state": args.state,
  "nonce": args.nonce,
  "redirect_uri": config["client"]["redirect_uri"],
  "response_type": "code id_token",
  "claims": {
    "id_token": {
      "mh:con_id": {
        "essential": True
      },
      "mh:payment": {
        "essential": True,
        "value": {
          "amount": 100,
          "payeeRef": "reference",
          "payerRef": "reference",
          "payeeId": payee_id,
        }
      }
    }
  },
  "aud": "{}/oidc".format(identity_server),
  "iss": client_id,
  **utils.get_common_claims()
}

request_uri = utils.post_request_object(config, params)
print("{}/oidc/auth?request_uri={}".format(identity_server, request_uri))